tests/conftest.py

```diff
@@ -1,4 +1,24 @@
 import sys
 from pathlib import Path
+from unittest.mock import patch
+
+import pytest
 
 sys.path.insert(0, str(Path(__file__).parent.parent))
+
+
+@pytest.fixture(scope="function")
+def generated_rows(request):
+    p = patch("snowfakery.output_streams.DebugOutputStream.write_single_row")
+    mockobj = p.start()
+
+    def fin():
+        p.stop()
+
+    def row_values(index, value):
+        return mockobj.mock_calls[index][1][1][value]
+
+    request.addfinalizer(fin)
+
+    mockobj.row_values = row_values
+    return mockobj
```

Comment:
You know you can implement pytest fixtures as a context manager, yield the fixture value, and then do cleanup after the yield statement, right?